src/app/analytics/analytics.factory.js

```diff
@@ -14,7 +14,7 @@ function analyticsFactory($window, $timeout, sessionService) {
         // Instantiate cart data layer
         $window.digitalData.cart = {
           id: cartData.id,
-          hash: cartData.id ? md5(cartData.id).toString() : null,
+          hash: cartData.id ? md5(cartData.id).toString().substring(0, 20) : null,
```

Comment:
How about `sha3(cartData.id, { outputLength: 80 }).toString()`? I feel like you might have more hash collisions if you truncate md5. I was looking at https://crypto.stackexchange.com/questions/3558/are-there-hash-algorithms-with-variable-length-output which suggested SHA3. `outputLength` specifies bits so 80 bits * 1 hex char / 4 bits = 20 hex chars.

https://www.webpackbin.com/bins/-KlykkvDQVIMFh_3DQgS